pkg/utils/httpClient.go

```diff
@@ -23,6 +23,7 @@ import (
 // NewRateLimitedClient returns a new HTTP client with rate limiter.
 func NewRateLimitedClient(qps int, burst int) (*http.Client, error) {
 	if qps == 0 {
+		fmt.Printf("Creating a default rate limited http client with QPS = %d, Burst = %d\n", qps, burst)
```

Comment:
This should be logged instead of printf

Reply:
Thanks for pointing this out. This is no longer needed since we are setting the QPS limit instead of leaving them as 0. I am pushing a new commit removing this.